src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -149,12 +148,37 @@ def right_size(
 
         self._init_sagemaker_session_if_does_not_exist()
 
+        self.temp_model_name = None
+        if isinstance(self, sagemaker.model.Model) and not isinstance(
+            self, sagemaker.model.ModelPackage
+        ):
+
+            unique_tail = uuid.uuid4()
+            self.temp_model_name = "SMPYTHONSDK-" + str(unique_tail)
+
+            create_model_args = dict(
+                name=self.temp_model_name,
+                role=self.role,
+                container_defs=None,
+                primary_container=self.prepare_container_def(),
+                vpc_config=self.vpc_config,
+                enable_network_isolation=self.enable_network_isolation(),
+            )
+            print(
+                f"Creating temporary model with name: {self.temp_model_name}"
+                " for Inference Recommender.",
+                flush=True,
+            )
+            self.sagemaker_session.create_model(**create_model_args)
```

Comment:
could we not just replace this whole block with this? 

```
self._create_sagemaker_model()
```

Do we need to create a temporary model name?

Can we not just use original name?

Reply:
1. [self._create_sagemaker_model()](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/model.py#L1640) set `container_defs` but we need to set `primary_container` 
2. using original model introduces edge case like
customer might already created a model with `original name` before using Python SDK, and that model could incorporate total different `image` and `modelDataUrl` than the `Model object` customer just created by Python SDK. Thus, using original model name need check and validation on this case.
We need to make a decision here if we need to use original name or create temp model anyway to invoke IR job. @jbarz1